infrastructure/microvm/cloudhypervisor/create.go

```diff
@@ -0,0 +1,196 @@
+	if err != nil {
+		return nil, fmt.Errorf("opening sterr file %s: %w", state.StderrPath(), err)
+	}
+
+	cmd.Stderr = stdErrFile
+	cmd.Stdout = stdOutFile
+	cmd.Stdin = &bytes.Buffer{}
+
+	var startErr error
+	if detached {
+		startErr = process.DetachedStart(cmd)
+	} else {
+		startErr = cmd.Start()
+	}
+
+	if startErr != nil {
+		return nil, fmt.Errorf("starting cloud hypervisor process: %w", err)
+	}
+
+	return cmd.Process, nil
+}
+
+func (p *provider) buildArgs(vm *models.MicroVM, state State) ([]string, error) {
+	args := []string{
+		"--api-socket",
+		state.SockPath(),
+		"--log-file",
+		state.LogPath(),
+		"-v",
+	}
+
+	// Kernel and cmdline args
+	kernelCmdLine := DefaultKernelCmdLine()
+
+	for key, value := range vm.Spec.Kernel.CmdLine {
+		kernelCmdLine.Set(key, value)
+	}
+
+	args = append(args, "--cmdline", kernelCmdLine.String())
+	args = append(args, "--kernel", fmt.Sprintf("%s/%s", vm.Status.KernelMount.Source, vm.Spec.Kernel.Filename))
+
+	// CPU and memory
+	args = append(args, "--cpus", fmt.Sprintf("boot=%d", vm.Spec.VCPU))
+	args = append(args, "--memory", fmt.Sprintf("size=%dM", vm.Spec.MemoryInMb))
+
+	// Volumes (root, additional, metadata)
+	rootVolumeStatus, volumeStatusFound := vm.Status.Volumes[vm.Spec.RootVolume.ID]
+	if !volumeStatusFound {
+		return nil, cerrors.NewVolumeNotMounted(vm.Spec.RootVolume.ID)
+	}
+	args = append(args, "--disk", fmt.Sprintf("path=%s", rootVolumeStatus.Mount.Source))
+	args = append(args, fmt.Sprintf("path=%s,readonly=on", state.CloudInitImage()))
+
+	for _, vol := range vm.Spec.AdditionalVolumes {
+		status, ok := vm.Status.Volumes[vol.ID]
+		if !ok {
+			return nil, cerrors.NewVolumeNotMounted(vol.ID)
+		}
+		args = append(args, fmt.Sprintf("path=%s", status.Mount.Source))
+	}
+
+	// Network interfaces
+	for i := range vm.Spec.NetworkInterfaces {
+		iface := vm.Spec.NetworkInterfaces[i]
+
+		status, ok := vm.Status.NetworkInterfaces[iface.GuestDeviceName]
+		if !ok {
+			return nil, cerrors.NewNetworkInterfaceStatusMissing(iface.GuestDeviceName)
+		}
+
+		args = append(args, "--net")
+		if iface.Type == models.IfaceTypeMacvtap {
+			arg, err := p.createMacvtapArg(&iface, status)
+			if err != nil {
+				return nil, fmt.Errorf("creating macvtap interface: %w", err)
+			}
+			args = append(args, arg)
+		} else if iface.Type == models.IfaceTypeTap {
+			args = append(args, fmt.Sprintf("tap=%s,mac=%s", status.HostDeviceName, iface.GuestMAC))
+		}
+	}
+
+	return args, nil
+}
+
+func (p *provider) createMacvtapArg(netInt *models.NetworkInterface, status *models.NetworkInterfaceStatus) (string, error) {
+	hostDevName := fmt.Sprintf("/dev/tap%d", status.Index)
+	fd, err := syscall.Open(hostDevName, syscall.O_RDWR, 755)
+	if err != nil {
+		return "", fmt.Errorf("getting file description for %s: %w", hostDevName, err)
+	}
+
+	arg := fmt.Sprintf("fd=%d,mac=", fd)
+	if netInt.GuestMAC != "" {
+		arg = arg + netInt.GuestMAC
+	}
```

Comment:
A thing to note here is that we are not defaulting the mac address to the host mac in the case that `GuestMac` is not set. This is what we do in firecracker: https://github.com/weaveworks-liquidmetal/flintlock/blob/main/infrastructure/firecracker/config.go#L184-L192

Do we want to maintain similar behaviour here?

Reply:
I need reacquaint myself why i did this, i vaguely remember there was a reason for changing this